adding new color property in Text and Link component

tejasnikam opened the pull request:
Fix issues [#2286 ](https://github.com/jpmorganchase/salt-ds/issues/2286)

joshwooding commented on packages/core/src/text/Text.tsx:
```diff
@@ -37,6 +37,7 @@ export type TextProps<T extends ElementType> = PolymorphicComponentPropWithRef<
      * Change text color palette
      */
     variant?: "primary" | "secondary";
+    color?: "primary" | "secondary";
```
We should copy Icon here

https://github.com/jpmorganchase/salt-ds/blob/5e48416c048e474b3750f405d677b608853b10ef/packages/icons/src/icon/Icon.tsx#L20

tejasnikam replied:
Needful has been done

joshwooding reviewed (changes requested):
Thanks for this, we need to do a bit more work here:

- Add two Storybook stories one for Link and Text showcasing the prop
- Update our QA stories to prevent visual regressions

There's some other steps too but we can tackle those after

tejasnikam commented:
> Thanks for this, we need to do a bit more work here:
> 
> * Add two Storybook stories one for Link and Text showcasing the prop
> * Update our QA stories to prevent visual regressions
> 
> There's some other steps too but we can tackle those after

Thanks for the quick review

-  Have added two Storybook stories one for Link and Text showcasing the prop
-  Updated QA stories to prevent visual regressions
-  Also updated documentation

tejasnikam commented:
Hello Team,
Any updates on this.
